diag_manager/fms_diag_axis_object.F90

```diff
@@ -275,6 +286,100 @@ subroutine set_axis_domain(obj, Domain, Domain2, DomainU)
     end select
   end subroutine set_axis_domain
 
+  subroutine fms_diag_axis_object_init()
+
+    if (module_is_initialized) return
+
+    number_of_axis = 0
+    allocate(axis_obj(max_axes))
+
+    module_is_initialized = .true.
+  end subroutine fms_diag_axis_object_init
+
+  subroutine fms_diag_axis_object_end()
+    deallocate(axis_obj)
+
+    module_is_initialized = .false.
+  end subroutine fms_diag_axis_object_end
+
+  !> @brief Wrapper for the register_diag_axis subroutine. This is needed to keep the diag_axis_init
+  !! interface the same
+  !> @return Axis id
+  FUNCTION modern_diag_axis_init(axis_name, axis_data, units, cart_name, long_name, direction,&
```

Comment:
In diag_axis_mod inside the original `diag_axis_init` is, I have

```
    if (use_modern_diag) then
      diag_axis_init = modern_diag_axis_init(name, DATA, units, cart_name, long_name=long_name, direction=direction,&
       & set_name=set_name, edges=edges, Domain=Domain, Domain2=Domain2, DomainU=DomainU, aux=aux, req=req, &
       & tile_count=tile_count, domain_position=domain_position )
      return
    endif
```
We can't move this to diag_manager because models _can have_ use `diag_axis_mod, only: diag_axis_init`.

Once we are ready to get rid of the old stuff, we can simply remove all the code below the if statement. We can probably even just the remove all the diag_axis_init and just have
```F90
USE fms_diag_axis_object_mod, ONLY: diag_axis_init => modern_diag_axis_init
```
Naming `modern_diag_axis_init`, `diag_axis_init` is confusing because now you will have two functions in two seperate modules with the same name. The original one is even made public by diag_manager_mod ...

`fms_diag_axis_init` is not descriptive. 

`modern_diag_axis_init` is best option, it is the "modern" version of diag_axis_init

Reply:
I think your approach in diag_axis_init with the if statement is a good idea.

Reply:
It is "modern" relative to the old diag_axis_init. We should clearly define the old and modern versions. If you name it "fms_diag_axis_object_init" then it is not clear. 

Once the old code is code is gone we can rename it to diag_axis_init and just have one.

Reply:
It should be named fms_diag_axis_init anyway because that's the direction/plan for public  FMS functions and routines. Using "modern" or "new" is not good practice. This is the kind of thing that has a plan for removal, then something happens and the routine is forever left with the "modern" prefix. How many namelist variables in our code reproduce some relatively ancient bug answers?